src/sql/mir.rs

```diff
@@ -964,6 +1122,34 @@ impl SqlToMirConverter {
                         })
                         .collect();
 
+                    // move predicates above grouped_by nodes
+                    for ccol in &computed_cols_cgn.columns {
+                        let over_col = target_columns_from_computed_column(ccol);
+                        let over_table = over_col.table.as_ref().unwrap().as_str();
+
+                        if pred_cols.contains_key(&over_col) {
+                            let parent = match prev_node {
+                                Some(p) => p,
+                                None => base_nodes[over_table].clone()
+                            };
+
+                            let new_mpns = self.reorder_predicates(
+                                    &format!(
+                                        "q_{:x}_n{}",
+                                        qg.signature().hash,
+                                        new_node_count
+                                    ),
+                                    &pred_cols,
+                                    over_col.clone(),
+                                    parent,
+                                    &mut moved_predicates);
+
```

Comment:
If I read the code correctly, `reorder_predicates` ends up generating nodes for *each* predicate associated with the `over_col` in `pred_cols`. Won't this end up generating nodes for the same predicate several times over? (since a predicate may be associated with multiple columns in `pred_cols`)